Src/FluentAssertions/Primitives/TimesConstraint.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using FluentAssertions.Common;
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Primitives
+{
+    public abstract class TimesConstraint
```

Comment:
I think we should keep the type `internal` for now. 

Also, since it's not a _verb_, I think the word _times_ is a bit weird. Maybe we should name this `OccurrenceConstraint`.

This class needs some documentation love.

Reply:
🤔 I agree on the naming, but the type cannot be marked `internal` since it's part of the public API https://github.com/fluentassertions/fluentassertions/pull/1145/files#diff-1ceb6d097a48e71dff2ed5ad377a1009R658

Reply:
Oh, you're right. In that case I think we should introduce an interface that only contains that single Assert method. The rest can be hidden from the public contract.